feat: dev서버에 에러가 났을 시 Slack으로 알람을 보낸다.

Longstick0 opened the pull request:
## 구현기능
EC2 dev 서버에 에러 발생 시 Slack으로 알람이 발생합니다.

### 세부 구현기능
- [x] 서버에서 에러가 났을 때 slack으로 에러 명시와 로그 레벨, 로그 출력
- [x] dev yml 파일 업데이트

### 관련 이슈
https://github.com/shophubkr/shophub-server/issues/5 dev

### 예시
![image](https://github.com/shophubkr/shophub-server/assets/69510442/b5ea257d-8121-4910-82f5-bf97f307935b)
하지만 필요 없는 정보까지 모두 담기는 거 같아서 현재는 아래와 같이 시간, 서버IP, 메세지명 정도만으로 줄였습니다.
```xml
<pattern>
        ${service} %level !!
        timestamp : "%d{yyyy-MM-dd'T'HH:mm:ss.SSSZ}" \n\n
        serverIp : "${java.rmi.server.hostname:-127.0.0.1}" \n\n
        hostName : "${nd.hostname:-localhost}" \n\n
        rest : "%msg" \n
</pattern>

Longstick0 committed:
feat: 서버에서의 에러 로그를 슬랙으로 전송하는 기능 구현

Longstick0 committed:
refactor: application-local.yml에 logging 삭제 후 EC2DEV환경에서 동작하도록 수정

Longstick0 committed:
refactor: 메세지에서 필요없는 부분 삭제

Longstick0 committed:
refactor: test용으로 사용했던 SecurityConfig.kt 제거

AnTaeho commented:
기능 자체는 좋은데 실제 슬랙에 올라오는 로그로는 정확히 무슨 상황인지 파악하기 힘들 것 같습니다.
다음과 같이 표현되도록 수정해보는건 어떨까요? 
링크 : https://mr-popo.tistory.com/229

![](https://img1.daumcdn.net/thumb/R1280x0/?scode=mtistory2&fname=https%3A%2F%2Fblog.kakaocdn.net%2Fdn%2FrpdZn%2Fbtr4vJG1UrP%2FGZFvaqxpFNqViIJCdfkV6K%2Fimg.png)

sectionr0 reviewed (approved):
고생하셨습니다👍

Longstick0 committed:
feat: 기존 XML을 통한 Slack Error 알람 기능을 AOP를 통해 구현

기존에는 서버에서 발생하는 모든 에러를 Slack 채널로 보냈다면, AOP를 통해 구현하여 원하는 ExceptionHandler에 적용 가능.

Longstick0 committed:
refactor: sendSlackMessage의 역할을 분리 및 스레드 생성 개수 조절

Longstick0 commented:
> 기능 자체는 좋은데 실제 슬랙에 올라오는 로그로는 정확히 무슨 상황인지 파악하기 힘들 것 같습니다. 다음과 같이 표현되도록 수정해보는건 어떨까요? 링크 : https://mr-popo.tistory.com/229
> 
> ![](https://camo.githubusercontent.com/7863974e5eb007545b885ae490a52291d124f569abf8f7f340a4206b6125a1f9/68747470733a2f2f696d67312e6461756d63646e2e6e65742f7468756d622f523132383078302f3f73636f64653d6d746973746f72793226666e616d653d6874747073253341253246253246626c6f672e6b616b616f63646e2e6e6574253246646e2532467270645a6e25324662747234764a4731557250253246475a467661717870464e715669494a4364666b56364b253246696d672e706e67)

보내주신 내용 바탕으로 다시 만들어봤습니다. 이 PR은 닫고 새로 만들어서 열어두겠습니다~!
